internal.lua

```diff
@@ -63,28 +63,57 @@ function digiline:rules_link_anydir(output, input)
 	or     digiline:rules_link(input, output)
 end
 
-function digiline:transmit(pos, channel, msg, checked)
-	local checkedid = tostring(pos.x).."_"..tostring(pos.y).."_"..tostring(pos.z)
-	if checked[checkedid] then return end
-	checked[checkedid] = true
+local function queue_new()
+	return {nextRead = 1, nextWrite = 1}
+end
 
-	local node = minetest.get_node(pos)
-	local spec = digiline:getspec(node)
-	if not spec then return end
+local function queue_empty(queue)
+	return queue.nextRead == queue.nextWrite
+end
 
+local function queue_push(queue, object)
```

Comment:
I would prefer the name `queue_enqueue` since "push" commonly refers to a stack operation.